src/librustc_back/target/mod.rs

```diff
@@ -698,6 +707,10 @@ impl ToJson for Target {
         target_option_val!(max_atomic_width);
         target_option_val!(panic_strategy);
 
+        if self.options.max_atomic_width.to_string() != self.target_pointer_width {
```

Comment:
Woah can't we use `Option` instead of magic `0` treatment or similar? Granted, I'm arriving late to this issue so apologies if that has already been ruled out for good reason.